store/mongo/migration_2_0_0.go

```diff
@@ -0,0 +1,314 @@
+// Copyright 2023 Northern.tech AS
+//
+//	Licensed under the Apache License, Version 2.0 (the "License");
+//	you may not use this file except in compliance with the License.
+//	You may obtain a copy of the License at
+//
+//	    http://www.apache.org/licenses/LICENSE-2.0
+//
+//	Unless required by applicable law or agreed to in writing, software
+//	distributed under the License is distributed on an "AS IS" BASIS,
+//	WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//	See the License for the specific language governing permissions and
+//	limitations under the License.
+package mongo
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	"go.mongodb.org/mongo-driver/mongo"
+	mopts "go.mongodb.org/mongo-driver/mongo/options"
+
+	"github.com/mendersoftware/go-lib-micro/identity"
+	"github.com/mendersoftware/go-lib-micro/mongo/migrate"
+	"github.com/mendersoftware/go-lib-micro/mongo/oid"
+	mstorev1 "github.com/mendersoftware/go-lib-micro/store"
+	mstore "github.com/mendersoftware/go-lib-micro/store/v2"
+	"go.mongodb.org/mongo-driver/bson"
+)
+
+const (
+	findBatchSize = 255
+)
+
+type migration_2_0_0 struct {
+	ds  *DataStoreMongo
+	ctx context.Context
+}
+
+var DbDevicesCollectionIndices = []mongo.IndexModel{
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldID, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldID).
+			SetUnique(true),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldStatus, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldStatus),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldIDDataSha, Value: 1},
+		},
+		//nolint:staticcheck // SA1019
+		Options: mopts.Index().
+			SetName(strings.Join([]string{
+				mstore.FieldTenantID,
+				dbFieldIDDataSha,
+			}, "_")).
+			SetUnique(true).
+			SetBackground(false),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldStatus, Value: 1},
+			{Key: dbFieldID, Value: 1},
+		},
+		//nolint:staticcheck // SA1019
+		Options: mopts.Index().
+			SetName(strings.Join([]string{
+				mstore.FieldTenantID,
+				dbFieldStatus,
+				dbFieldID,
+			}, "_")).
+			SetBackground(false),
+	},
+}
+
+var DbAuthSetsCollectionIndices = []mongo.IndexModel{
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldID, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(mstore.FieldTenantID + "_" + dbFieldID).
+			SetUnique(true),
+	},
+	{
+		Keys: bson.D{
+			{Key: mstore.FieldTenantID, Value: 1},
+			{Key: dbFieldDeviceID, Value: 1},
+			{Key: dbFieldIDDataSha, Value: 1},
+			{Key: dbFieldPubKey, Value: 1},
+		},
+		Options: mopts.Index().
+			SetName(strings.Join([]string{
+				mstore.FieldTenantID,
+				dbFieldDeviceID,
+				dbFieldIDDataSha,
+				dbFieldPubKey,
+			}, "_")).
+			SetUnique(true),
+	},
```

Comment:
Can we please drop this index? The uniqueness constraint is already held by the index below. I think all we actually need to index is `{tenant_id: 1, device_id: 1}`
```suggestion
	{
		Keys: bson.D{
			{Key: mstore.FieldTenantID, Value: 1},
			{Key: dbFieldDeviceID, Value: 1},
		},
		Options: mopts.Index().
			SetName(strings.Join([]string{
				mstore.FieldTenantID,
				dbFieldDeviceID,
			}, "_"))
	},
```

Reply:
sure.